src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5664,7 +5664,8 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                 hasError = this.CheckInstanceOrStatic(node, receiver, fieldSymbol, ref resultKind, diagnostics);
             }
 
-            if (!hasError && fieldSymbol.IsFixed)
+            var notInsideNameOf = EnclosingNameofArgument == null;
+            if (!hasError && fieldSymbol.IsFixed && notInsideNameOf)
```

Comment:
Consider inlining `notInsideNameOf` so the check is only made if the other two expressions are also true.